### PR TITLE
Enhance individual UT in go-cose library to handle specific errors

### LINKED ISSUE #138: Further work: Unit Test in go-cose repository needs to check for precise error condition
This issue is an extension of already fixed issue #118 

Issue #118 fixed the specific cases in verifier_test.go.

However after tidying up in file: verifier_test.go realized that a similar change is missing from other parts of go-cose unit test files.

This change covers the set for all go-cose files

### PR DESCRIPTION
**Pull request by yogeshbdeshpande**
Fixes #138 

Signed-off-by: Yogesh Deshpande <yogesh.deshpande@arm.com>

### CODE REVIEW

**Review by thomas-fossati — APPROVED** [review]
thanks!

**Comment by yogeshbdeshpande** [comment]
@shizhMSFT and @qmuntal  
Request please review as well...

**Review by qmuntal — APPROVED** [review]
LGTM and nice!

**Comment by SteveLasker** [comment]
@yogeshbdeshpande, can you fix the signing?  [Learn more about signing commits.](https://docs.github.com/articles/about-gpg/)
@shizhMSFT, can you please review?

File: ecdsa_test.go [hunk 1]
```diff
@@ -98,11 +98,12 @@ func TestI2OSP(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			err := I2OSP(tt.x, tt.buf)
-			if (err != nil) != tt.wantErr {
+			if err != nil && (err.Error() != tt.wantErr) {
```
Comment (shizhMSFT) [inline ecdsa_test.go:101]: What if `err == nil` but `tt.wantErr` is non-empty?

Comment (yogeshbdeshpande) [reply ecdsa_test.go:101]: Done the changes in the recent commit

**Pull request merged**